catalog_tools/analysis/estimate_beta.py

```diff
@@ -230,6 +233,167 @@ def estimate_b_laplace(
+        std_b_parameter: standard error of b_parameter
+        rate_at_lmc: maximum likelihood point estimate of earthquake rate
+                     at the lower magnitude of completeness
+        std_rate_at_lmc: standard error of rate_at_lmc
+        a_val: maximum likelihood point estimate of a-value
+               ( =log10(rate at mag=0) ) of Gutenberg-Richter
+               magnitude frequency distribution
+    """
+    assert len(magnitudes) == len(years), \
+        "the magnitudes and years arrays have different lengths"
+    assert completeness_table.shape[1] == 2
+    assert np.all(np.ediff1d(completeness_table[:, 0]) >= 0),\
+        "magnitudes in completeness table not in ascending order"
+    assert [i - delta_m in np.arange(completeness_table[0, 0],
+                                     mag_max + 0.001, delta_m)
+            for i in np.unique(magnitudes)],\
+        "magnitude bins not aligned with completeness edges"
+    if not np.all(magnitudes >= completeness_table[:, 0].min()):
+        warnings.warn(
+            "magnitudes below %.2f are not covered by the "
+            "completeness table and are discarded" %
+            completeness_table[0, 0])
+    assert b_parameter == 'b_value' or b_parameter == 'beta', \
+        "please choose either 'b_value' or 'beta' as b_parameter"
+    factor = 1 / np.log(10) if b_parameter == 'b_value' else 1
+    tend = tend if tend else np.max(years)
+
+    # Get the magnitudes and completeness years as separate arrays
+    completeness_magnitudes = completeness_table[:, 0]
+    completeness_years = completeness_table[:, 1]
+
+    # Obtain the completeness start year for each value in magnitudes
+    insertion_indices = np.searchsorted(completeness_magnitudes, magnitudes)
+    cmpl_yrs = np.array(
+        [completeness_years[idx - 1] if idx not in [0, len(completeness_years)]
+         else {0: -1, len(completeness_years): completeness_years[-1]}[idx]
+         for i, idx in enumerate(insertion_indices)]
+    )
+
+    # filter out events outside completeness window and
+    # get number of "complete" events in each magnitude bin
+    # and associated year of completeness
+    idxcomp = ((cmpl_yrs > 0) & (years - cmpl_yrs >= 0))
+    complete_events = pd.DataFrame.groupby(
+        pd.DataFrame(data={
+            'mag_left_edge': np.array(
+                [i.left for i in pd.cut(magnitudes[idxcomp],
+                 bins=np.arange(completeness_magnitudes[0],
+                 mag_max + 0.01, delta_m), right=False)]),
+            'completeness_start': cmpl_yrs[idxcomp]
+        }), by=['mag_left_edge', 'completeness_start'])\
+        .size()\
+        .to_frame('num')\
+        .reset_index()
+    assert np.all(
+        complete_events.completeness_start > 0)  # should be the case by design
+
+    # minimization
+    beta = np.log(10)  # initialization of beta
+    output = minimize(weichert_minimization,
```

Comment:
Does the output depend on beta?

Reply:
It is just the initial guess from which the minimisation starts. It shouldn't depend on it, unless you put something very extreme